packages/client/src/containers/Datasets/DatasetList.tsx

```diff
@@ -0,0 +1,400 @@
+import { errorHandler } from 'utils/errorHandler';
+
+import {
+  GroupContext,
+  GroupContextComponentProps,
+  withGroupContext,
+} from 'context/group';
+
+import {
+  Dataset,
+  DatasetConnection,
+  InputVariables,
+  QueryVariables,
+} from 'components/datasets/common';
+import { DatasetCreateForm } from 'components/datasets/CreateForm';
+import {
+  CreateDatasetMutation,
+  DeleteDatasetMutation,
+  GetDatasets,
+} from './dataset.graphql';
+
+const { confirm } = Modal;
+
+const DEFAULT_PAGE = 1;
+const DEFAULT_PAGE_SIZE = 10;
+const Search = Input.Search;
+
+type Props = {
+  groups: Array<{
+    id: string;
+  }>;
+  datasets: {
+    error?: Error | undefined;
+    loading: boolean;
+    variables: QueryVariables;
+    refetch: (variables?: QueryVariables) => void;
+    datasetV2Connection: DatasetConnection;
+  };
+  createDataset: ({
+    variables,
+  }: {
+    variables: {
+      payload: InputVariables;
+    };
+  }) => Promise<void>;
+  deleteDataset: ({
+    variables,
+  }: {
+    variables: {
+      where: { id: string; groupName: string };
+    };
+  }) => Promise<void>;
+} & RouteComponentProps &
+  GroupContextComponentProps;
+
+function CommonPageTitle() {
+  const breadcrumbs = [
+    {
+      key: 'list',
+      matcher: /\/datasets/,
+      title: 'Datasets',
+      link: '/datasets',
+      tips: 'Users can manage datasets here.',
+      // TODO: add doc link
+      tipsLink: '',
+    },
+  ];
+
+  return <PageTitle breadcrumb={<Breadcrumbs pathList={breadcrumbs} />} />;
+}
+
+function _DatasetList({
+  groups,
+  datasets,
+  createDataset,
+  deleteDataset,
+}: Props) {
+  const groupContext = React.useContext(GroupContext);
+  const [keyword, setKeyword] = React.useState('');
+  const [modalVisible, setModalVisible] = React.useState(false);
+
+  async function onSubmit(data: InputVariables) {
+    const { refetch, variables } = datasets;
+
+    await createDataset({
+      variables: {
+        payload: {
+          ...data,
+          groupName: groupContext.name,
+        },
+      },
+    });
+
+    refetch({
+      where: variables.where,
+      page: variables.page,
+    });
+  }
+
+  function onPageChanged(page) {
+    const { refetch, variables } = datasets;
+
+    refetch({
+      where: variables.where,
+      page,
+    });
+  }
+
+  function searchHandler(keyword: string) {
+    const { refetch, variables } = datasets;
+
+    refetch({
+      where: {
+        ...variables.where,
+        search: keyword,
+      },
+      page: DEFAULT_PAGE,
+    });
+  }
+
+  function renderName(text, record) {
+    return (
+      <TruncateTableField text={text}>
+        <Link
+          to={{
+            state: {
+              prevPathname: location.pathname,
+              prevSearch: location.search,
+            },
+            pathname: `datasets/${record.id}`,
+          }}
+        >
+          {text}
+        </Link>
+      </TruncateTableField>
+    );
+  }
+
+  function renderTags(text, record) {
```

Comment:
Seems the text not be used, you can change it as:

```suggestion
  function renderTags(record) {
```

And don't forget to delete Line 275 (`dataIndex`).